src/ServicePulse.Host/Owin/Microsoft/StaticFileContext.cs

```diff
@@ -218,8 +218,7 @@ public Task SendAsync()
             var readStream = fileInfo.CreateReadStream();
             var copyOperation = new StreamCopyOperation(readStream, response.Body, length, request.CallCancelled);
             var task = copyOperation.Start();
-            _ = task.ContinueWith(resultTask => readStream.Close(), TaskContinuationOptions.ExecuteSynchronously);
-            return task;
+            return task.ContinueWith(resultTask => readStream.Close(), TaskContinuationOptions.ExecuteSynchronously);
```

Comment:
I think this is an unnecessary optimization that complicates things for now added value. I would just switch this to async-await style. That is more readable than continuations and is well aligned with https://github.com/davidfowl/AspNetCoreDiagnosticScenarios/blob/master/AsyncGuidance.md#prefer-await-over-continuewith

This code is anyway not exactly on the hot path as far as I'm concerned

Reply:
👍 I'll amend the commit to do that.